scst-sign/configuring.md

```diff
@@ -0,0 +1,244 @@
+the images.
+- A list of image name patterns against which the policy is enforced. Each image
+name pattern is mapped to the required public keys.
+- Optionally, a secret that grants authentication to a given name pattern in
+your policy.
+
+The following is an example `ClusterImagePolicy`:
+
+```yaml
+---
+apiVersion: signing.run.tanzu.vmware.com/v1alpha1
+kind: ClusterImagePolicy
+metadata:
+    name: image-policy
+spec:
+  verification:
+    exclude:
+      resources:
+        namespaces:
+        - kube-system
+    keys:
+    - name: first-key
+      publicKey: |
+        -----BEGIN PUBLIC KEY-----
+        ...
+        -----END PUBLIC KEY-----
+    images:
+    - namePattern: registry.example.org/myproject/*
+      keys:
+      - name: first-key
+```
+
+> **Note**:
+>   - The `name` for the `ClusterImagePolicy` resource must be `image-policy`.
+>   - Add any namespaces that run container images that are not signed in the
+>   `spec.verification.exclude.resources.namespaces` section, such as the
+>   `kube-system` namespace.
+>   - If no `ClusterImagePolicy` resource is created all images are admitted into
+>   the cluster with the following warning:
+>     ```
+>     Warning: clusterimagepolicies.signing.run.tanzu.vmware.com "image-policy" not found. Image policy enforcement was not applied.
+>     ```
+>   - For a simpler installation process in a non-production environment
+>   VMware recommends you use the manifest below to create the `ClusterImagePolicy`
+>   resource. This manifest includes a cosign public key which signed the public
+>   cosign v1.2.1 image. The cosign public key validates the specified cosign
+>   images. Container images running in system namespaces are currently not
+>   signed. You must configure the image policy webhook to allow these unsigned
+>   images by adding system namespaces to the
+>   `spec.verification.exclude.resources.namespaces` section.
+
+```bash
+cat <<EOF | kubectl apply -f -
+apiVersion: signing.run.tanzu.vmware.com/v1alpha1
+kind: ClusterImagePolicy
+metadata:
+  name: image-policy
+spec:
+  verification:
+    exclude:
+      resources:
+        namespaces:
+        - kube-system
+    keys:
+    - name: cosign-key
+      publicKey: |
+        -----BEGIN PUBLIC KEY-----
+        MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEhyQCx0E9wQWSFI9ULGwy3BuRklnt
+        IqozONbbdbqz11hlRJy9c7SG+hdcFl9jE9uE/dwtuwU2MqU9T/cN0YkWww==
+        -----END PUBLIC KEY-----
+    images:
+    - namePattern: gcr.io/projectsigstore/cosign*
+      keys:
+      - name: cosign-key
+EOF
+```
+
+## <a id='proving-credentials-for-the-webhook'></a> Providing credentials for the webhook
+
+There are four ways the webhook reads credentials to authenticate to registries
+protected by authentication, in order:
+1. Reading `imagePullSecrets` directly from the resource being admitted.
+1. Reading `imagePullSecrets` from the service account the resource is running
```

Comment:
How does would look in terms of configuration in my cluster ? Do I have to create any resource ?  I suggest to explain it in more details.

Reply:
Or either add a link to another section.

Reply:
Since this is a Kubernetes configuration, does it make sense to link to Kubernetes Secret documentation?

Reply:
yes, it is alright to point to the kubernetes image pull secrets documentation https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/